### DISCUSSION TITLE
Download from Git

### DISCUSSION THREAD

**Discussion by cybercop23** [xLightsSequencer/xLights#4501] [Ideas]
I'm working on switching the downloads from xlights.org to Git. I have it working where it checks the git releases list and can even set a nightly version or the current prod release (as well as set the appimage).
I'm also looking to allow users to choose what track they want to be on... Nightly, Beta or Production.

Questions:

1.  @dkulp Are there any issues/concerns with moving the downloads from xlights.org to git? Any issues with the increase number of requests and download/traffic bandwidth?
2. @computergeek1507 Can we set up a new tag, Beta, that builds the nighly only one day of the week, Thursday or Friday? Users can then be on the Beta track and only get it once/week vs daily if they are on the Nightly build.
3. Anything else that you think we should consider. Missing or overlooking anything?

**Comment by computergeek1507** [comment]
Are you changing the update check to download the installer and run it as well?

**Reply by cybercop23** [reply]
Yuppers. So once buit in Git, it will automagically be avail in xlights for check.

**Reply by derwin12** [reply]
Need to address the xlights.org file ... redirect/upload there whatever...

**Reply by cybercop23** [reply]
If we get the beta tag, we can point to that... actually.. for download we already have it...
https://github.com/xLightsSequencer/xLights/releases/latest

**Reply by cybercop23** [reply]
`2024-04-15 14:32:03,640 18516 log_base [DEBUG] Downloading releases from https://api.github.com/repos/xLightsSequencer/xLights/releases
2024-04-15 14:32:03,641 18516 log_base [DEBUG] File Cache downloading file https://api.github.com/repos/xLightsSequencer/xLights/releases.
2024-04-15 14:32:03,642 18516 log_base [DEBUG] Making request to 'https://api.github.com/repos/xLightsSequencer/xLights/releases' -> C:\Users\Alex\AppData\Local\Temp\xlEC1F..
2024-04-15 14:32:03,642 18516 log_base [DEBUG] https://api.github.com/repos/xLightsSequencer/xLights/releases
2024-04-15 14:32:04,110 18516 log_base [DEBUG] Current Version: '2024.08.1'. Latest Available '2024.08'. Skip Version ''.
2024-04-15 14:32:16,832 18516 log_base [DEBUG] User has chosen to upgrade to version 2024.08. URL: https://github.com/xLightsSequencer/xLights/releases/download/2024.08/xLights64_2024_8.exe`

**Comment by derwin12** [comment]
Do we need the 2 digit value _08 so that it can tell its newer .. or is it based on date?

**Reply by cybercop23** [reply]
Does it in parts, but the version comp was always using the 2 digits and that's what I pull from git.

`wxArrayString compare_parts = wxSplit(compare, '.');
wxArrayString version_parts = wxSplit(version, '.');
if (wxAtoi(version_parts[0]) < wxAtoi(compare_parts[0])) return true;
if (wxAtoi(version_parts[0]) > wxAtoi(compare_parts[0])) return false;
if (wxAtoi(version_parts[1]) < wxAtoi(compare_parts[1])) return true;
if (wxAtoi(version_parts[1]) > wxAtoi(compare_parts[1])) return false;`

  "tag_name": "2024.08",

**Comment by keithsw1111** [comment]
Last I was aware the nightly builds were missing some things ... SMPTE maybe?

**Comment by computergeek1507** [comment]
All xSchedule builds on GitHub are missing SMPTE support as I could not get the code to build locally or on GitHub. It causes linker errors when a new version of msbuild is used and the library objects are not compatible. It could be built without optimization to prevent the issue with different linkers versions as well.

**Reply by cybercop23** [reply]
So it doesn't seem like we can move to getting the latest version from github, until we fix this issue.
Is is fixable? And should we work to make the move, or just keep it to nighly and weekly betas?

**Reply by computergeek1507** [reply]
If Keith uploads a fork of his SMPTE library to the xLights Sequencer github I can look into fixing it.

**Reply by keithsw1111** [reply]
I can ... just not until i get back home ... so a week or so.

**Reply by cybercop23** [reply]
Hi @keithsw1111. Hope you're feeling better. Wheneven you get a min, can you please take a look at this?
